lib/prx_auth/rails/ext/controller.rb

```diff
@@ -26,6 +29,16 @@ def current_user
         PrxAuth::Rails::Token.new(prx_auth_token)
       end
 
+      def account_name_for(id)
+        return "Unknown #{id}" unless session['prx.account.name.mapping'].present?
```

Comment:
If the ID is missing, could this just-in-time fetch it?  That would allow fetching names for IDs not in your token and stale tokens.  Also prevent re-fetching account_ids ID isn't returning, by setting `session[key][id] = nil` or something.

(Also, should probably use that PRX_ACCOUNT_NAME_MAPPING_KEY constant here).

Reply:
👍